### ISSUE TITLE
Icons in geojson source randomly vanish depending on zoom level / tile

### ISSUE THREAD

**Issue by westnordost** [maplibre/maplibre-native#2372]
Some icons in a symbol layer from a geojson source are not shown at certain zoom levels, while their text field, if they have one, is always shown.
Changing the zoom level (up or down) may eventually make them appear and/or vanish again.

The style used for the layer (in code) is

```kotlin
iconImage(get("icon")),
textField(get("label")),
textAnchor(Property.TEXT_ANCHOR_TOP),
textOffset(literal(arrayOf(0f, 1f))),
iconAllowOverlap(true),
textAllowOverlap(true)
```

Whether the geojson source is set to `isVolatile` or not does not change whether the issue appears or not.

Due to `iconAllowOverlap(true)`, the icon should always show, just like the text, independent of zoom level or presence of any other icons or text. However, for the tests in the videos below, I still removed all layers from the map except for these icons and some buildings and landuse for orientation - this is why it looks so empty. Most importantly, there is no other symbol layer. I also removed all properties of this symbol layer that were irrelevant for reproducing the issue. 

**Exhibit 1**: Some icons vanish and reappear seemingly depending on the zoom level. However, the texts never vanishes

https://github.com/maplibre/maplibre-native/assets/4661658/e2371c1a-4c7b-4cb5-9698-4e0144fbbf58

**Exhibit 2**: Here we have a high density of icons that never have text. Due to the density of the icons displayed here, it becomes apparent that whether or not an icon is displayed **might be related to in which tile it is located**, since the icons always vanish or appear in square-ish-looking groups.

https://github.com/maplibre/maplibre-native/assets/4661658/db9992cf-3721-49f2-954e-dc38ead87eda

### Platform information
 - OS: Android API 34 emulator (Resizable (Experimental)), x86_64, no google play services
 - MapLibre Version Android 11.0-pre6

### Reproduction information

So far, I personally have only been able to reproduce this issue in an emulator.

However, @Helium314 did describe a similar (or the same?) issue on his phone. @Helium314: Is this what you are seeing on your phone? (If yes, would you add the device+OS version?) Or is it another issue?

**Comment by Helium314** [comment]
> However, @Helium314 did describe a similar (or the same?) issue on his phone. @Helium314: Is this what you are seeing on your phone? (If yes, would you add the device+OS version?) Or is it another issue?

I'm not sure, and I can't reproduce it reliably. It was similar, but the showing / not showing icons were the same for all zoom levels.
I had a similar issue with overlays in general (same source as your icons), but also not reproducible.
Interestingly both issues were permanently occurring when I first encountered them, but vanished after unistalling and re-installing the app. After that I managed to get it only once, I think.
The app still filled the source and added the layers (I checked that). Is there anything possibly related MapLibre stores on disk besides the offline tile data (which should not be related)?

**Comment by westnordost** [comment]
I, too, observed the issue you describe on my phone once or twice, but I could never reproduce it after the first "freak" occurrence, while this issue I can reliably reproduce but only on emulators.

I can only hope that that issue is the same or caused by the same thing as this issue as this one, because it'd be a nightmare to reproduce, while at the same time at least for our use case (OpenStreetMap editor), it is a critical issue - POIs not displayed on the map will be added twice (thrice, ...) by users, thinking they are missing.

**Comment by kocheick** [comment]
I doubt this issue is related to the source

I started noticing the same issue on a Compose project, thinking it was related to the icons loading, after updating to maplibre 10.x.x ~ and latest compose updates. It's as if the updates brought this issue because it wasn't happening with the previous versions but the previous versions were buggy, although many of those bugs were fixed this new one came out of it

**Comment by westnordost** [comment]
You updated to 10.x or 11.x?

**Comment by kocheick** [comment]
Went from 9.x.x to 10.x.x

**Comment by louwers** [comment]
If something is not reproducible on a real device, it's likely an emulator bug and it should be reported [upstream](https://developer.android.com/studio/report-bugs#emulator-bugs).

Feel free to re-open if this also shows up on real devices.

**Comment by kocheick** [comment]
I tested it in the emulator and now all is well after the update to maplibre 11, and android plugins to 3.0.0
